### ISSUE TITLE
issue with cloudflare-dns-server in mac (docker)

### ISSUE THREAD

**Issue by gmartinerm** [qdm12/cloudflare-dns-server#7]
I spoke too soon last time I opened my last issue.  Let me describe my setup: I have two hosts

192.168.1.2 (pihole-brutus) running pihole and cloudflare-dns.  The upstream DNS servers for this host are 127.0.0.1#53 and 192.168.1.3#5053

192.168.1.3 (pihole-suffix) running pihole and cloudflare-dns. The upstream DNS servers for this host are 127.0.0.1#53 and 192.168.1.2#5053

If I point one of my local machines to 192.168.1.2 as its resolver the name resolution works.  I can specify individually either one of their DNS upstreams in pihole (or both at the same time) and name resolution works.

If I point one of my local machines to 192.168.1.3 as its resolver the name resolution doesn't work.  If I modify the upstream DNS servers and I used anything other than localhost:5053 as the upstream DNS for this pihole the name resolution works.

```
suffix:~ admin$ docker ps
CONTAINER ID        IMAGE                          COMMAND                  CREATED             STATUS                    PORTS                                                                                      NAMES
ce8c47910ae1        qmcgaw/cloudflare-dns-server   "/bin/sh -c /etc/unb…"   11 minutes ago      Up 11 minutes (healthy)   0.0.0.0:5053->53/udp                                                                       cloudflare-suffix
ce1c373db763        pihole/pihole:latest           "/s6-init"               13 minutes ago      Up 13 minutes (healthy)   0.0.0.0:53->53/tcp, 0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp, 0.0.0.0:53->53/udp, 67/udp   pihole-suffix
```

docker logs for 192.168.1.3
```
suffix:~ admin$ docker logs ce8c47910ae1
 =========================================
 =========================================
 === CLOUDFLARE DNS OVER TLS CONTAINER ===
 =========================================
 =========================================
 == by github.com/qdm12 - Quentin McGaw ==

Running as root
Unbound version: 1.8.3
Unbound DNS server: google
Unbound listening UDP port: 53
Verbosity level set to 1 on 5
Verbosity details level set to 0 on 4
Malicious hostnames and ips blocking is on
147173 malicious hostnames and 16021 malicious IP addresses blacklisted
[1554347234] unbound[35:0] notice: init module 0: validator
[1554347234] unbound[35:0] notice: init module 1: iterator
[1554347234] unbound[35:0] info: start of service (unbound 1.8.3).
[1554347532] unbound[35:1] info: generate keytag query _ta-4a5c-4f66. NULL IN
suffix:~ admin$
```
I almost forgot to mention 192.168.1.3 is a a mac mini running docker for OSX while 192.168.1.2 is Synology NAS running docker.

Now here is the docker inspect output
```
suffix:~ admin$ docker inspect cloudflare-suffix
[
    {
        "Id": "ce8c47910ae14a97bcc625e7c677e2fc92c8c8788a3fb551efc2a853940ec4e3",
        "Created": "2019-04-04T03:07:11.340120062Z",
        "Path": "/bin/sh",
        "Args": [
            "-c",
            "/etc/unbound/entrypoint.sh"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 5091,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2019-04-04T03:07:12.251347814Z",
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Health": {
                "Status": "healthy",
                "FailingStreak": 0,
                "Log": [
                    {
                        "Start": "2019-04-04T03:12:12.034163831Z",
                        "End": "2019-04-04T03:12:12.85962567Z",
                        "ExitCode": 0,
                        "Output": ""
                    },
                    {
                        "Start": "2019-04-04T03:17:12.651246179Z",
                        "End": "2019-04-04T03:17:12.76505059Z",
                        "ExitCode": 0,
                        "Output": ""
                    },
                    {
                        "Start": "2019-04-04T03:22:12.562023379Z",
                        "End": "2019-04-04T03:22:12.682359992Z",
                        "ExitCode": 0,
                        "Output": ""
                    },
                    {
                        "Start": "2019-04-04T03:27:12.495949075Z",
                        "End": "2019-04-04T03:27:12.595405771Z",
                        "ExitCode": 0,
                        "Output": ""
                    }
                ]
            }
        },
        "Image": "sha256:074deaa4492620f377df84b174042365e21b1d558df49498826722e0b3164b6b",
        "ResolvConfPath": "/var/lib/docker/containers/ce8c47910ae14a97bcc625e7c677e2fc92c8c8788a3fb551efc2a853940ec4e3/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/ce8c47910ae14a97bcc625e7c677e2fc92c8c8788a3fb551efc2a853940ec4e3/hostname",
        "HostsPath": "/var/lib/docker/containers/ce8c47910ae14a97bcc625e7c677e2fc92c8c8788a3fb551efc2a853940ec4e3/hosts",
        "LogPath": "/var/lib/docker/containers/ce8c47910ae14a97bcc625e7c677e2fc92c8c8788a3fb551efc2a853940ec4e3/ce8c47910ae14a97bcc625e7c677e2fc92c8c8788a3fb551efc2a853940ec4e3-json.log",
        "Name": "/cloudflare-suffix",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": null,
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {
                "53/udp": [
                    {
                        "HostIp": "",
                        "HostPort": "5053"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "no",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "shareable",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DiskQuota": 0,
            "KernelMemory": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": 0,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/4ab249cbbc0a1648d8dec74313fb45d9cea97f4202f1851b9304979b7d55f5f4-init/diff:/var/lib/docker/overlay2/3987f2517fb0b8091bf9025c19a72320918ebb39faa42df5002eabaadd00500b/diff:/var/lib/docker/overlay2/33b5039a9653bfd714652763b6a01e28e335603a44059cd6541b5995a8a5f218/diff:/var/lib/docker/overlay2/3ab1aeeb147eec6a1f4b114d7500e0e9b75cb2167f5cb208f70f61279b10a51b/diff:/var/lib/docker/overlay2/1570a4b8101ee5de4051f8321ee64012b4975d429a20f0682b0493a48e1f0322/diff",
                "MergedDir": "/var/lib/docker/overlay2/4ab249cbbc0a1648d8dec74313fb45d9cea97f4202f1851b9304979b7d55f5f4/merged",
                "UpperDir": "/var/lib/docker/overlay2/4ab249cbbc0a1648d8dec74313fb45d9cea97f4202f1851b9304979b7d55f5f4/diff",
                "WorkDir": "/var/lib/docker/overlay2/4ab249cbbc0a1648d8dec74313fb45d9cea97f4202f1851b9304979b7d55f5f4/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [],
        "Config": {
            "Hostname": "a7680fbfe37a",
            "Domainname": "",
            "User": "root",
            "AttachStdin": false,
            "AttachStdout": true,
            "AttachStderr": true,
            "ExposedPorts": {
                "53/udp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PROVIDER=google",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "VERBOSITY=1",
                "VERBOSITY_DETAILS=0",
                "BLOCK_MALICIOUS=on",
                "LISTENINGPORT=53"
            ],
            "Cmd": null,
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "LISTENINGPORT=${LISTENINGPORT:-53}; [ -z $(nslookup duckduckgo.com 127.0.0.1 -port=$LISTENING_PORT -timeout=1 | grep \"no servers could be reached\") ] || exit 1"
                ],
                "Interval": 300000000000,
                "Timeout": 15000000000,
                "StartPeriod": 5000000000,
                "Retries": 1
            },
            "ArgsEscaped": true,
            "Image": "qmcgaw/cloudflare-dns-server",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/bin/sh",
                "-c",
                "/etc/unbound/entrypoint.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "cpu-usage": "Low",
                "image-size": "17MB",
                "maintainer": "quentin.mcgaw@gmail.com",
                "org.label-schema.build-date": "2019-04-02T06:27:35Z",
                "org.label-schema.docker.cmd": "docker run -d -p 53:53/udp qmcgaw/cloudflare-dns-server",
                "org.label-schema.docker.cmd.devel": "docker run -it --rm -p 53:53/udp -e VERBOSITY=3 -e VERBOSITY_DETAILS=3 -e BLOCK_MALICIOUS=off qmcgaw/cloudflare-dns-server",
                "org.label-schema.docker.params": "VERBOSITY=from 0 (no log) to 5 (full debug log) and defaults to 1,VERBOSITY_DETAILS=0 to 4 and defaults to 0 (higher means more details),BLOCK_MALICIOUS='on' or 'off' and defaults to 'on' (note that it consumes about 50MB of additional RAM),LISTENING_PORT=1 to 65535 for internal Unbound listening port,PROVIDER=CLOUDFLARE or GOOGLE or QUAD9 or QUADRANT or CLEANBROWSING",
                "org.label-schema.schema-version": "1.0.0-rc1",
                "org.label-schema.url": "https://github.com/qdm12/cloudflare-dns-server",
                "org.label-schema.vcs-description": "Runs a local DNS server connected to Cloudflare DNS server 1.1.1.1 over TLS (and more)",
                "org.label-schema.vcs-ref": "98fc026",
                "org.label-schema.vcs-url": "https://github.com/qdm12/cloudflare-dns-server",
                "org.label-schema.vcs-usage": "https://github.com/qdm12/cloudflare-dns-server/blob/master/README.md#setup",
                "ram-usage": "13.2MB to 70MB"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "86211990d33959b1181f3c662851b32507eacaf26d1ba0d0a35336889235dddf",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "53/udp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "5053"
                    }
                ]
            },
            "SandboxKey": "/var/run/docker/netns/86211990d339",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "bf16c767ca6eb97dceaf0aa685c4ce4e69237dfe917ab774c20378f9152ecbdb",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.3",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:03",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [],
                    "NetworkID": "25e5cea229aed6ccfaeae322475a357279337582a3b0149504cfb9176eb1c7e0",
                    "EndpointID": "bf16c767ca6eb97dceaf0aa685c4ce4e69237dfe917ab774c20378f9152ecbdb",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.3",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:11:00:03",
                    "DriverOpts": null
                }
            }
        }
    }
]
```
The healthcheck for the container passes
```
suffix:~ admin$ docker exec cloudflare-suffix nslookup duckduckgo.com 127.0.0.1 -port=53 -timeout=1
Server:         127.0.0.1
Address:        127.0.0.1#53

Non-authoritative answer:
Name:   duckduckgo.com
Address: 23.21.193.169
Name:   duckduckgo.com
Address: 107.20.240.232
Name:   duckduckgo.com
Address: 184.72.104.138

suffix:~ admin$
```
It is almost like the pihole instance in OSX can't communicate with cloudflare-dns-server over port 5053 right on the same host.  Here is the output of netstat for UDP port 5053

```
bash-3.2# netstat -at | grep 5053
udp4       0      0  *.5053                 *.*
```

Really appreciate your support. I spent at least 3 hours trying to debug this problem. Happy to arrange whatever access or whatever information you need

**Comment by gmartinerm** [comment]
Adding the docker inspect for the pihole container
```
suffix:~ admin$ docker inspect pihole-suffix
[
    {
        "Id": "ce1c373db763d38d872811cab3ee5cdec1aabe50199f44ac043862ae3e34c4b1",
        "Created": "2019-04-04T03:05:31.521639043Z",
        "Path": "/s6-init",
        "Args": [],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 4017,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2019-04-04T03:05:32.465998535Z",
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Health": {
                "Status": "healthy",
                "FailingStreak": 0,
                "Log": [
                    {
                        "Start": "2019-04-04T04:06:15.291415366Z",
                        "End": "2019-04-04T04:06:15.43591286Z",
                        "ExitCode": 0,
                        "Output": "\n; <<>> DiG 9.10.3-P4-Debian <<>> @127.0.0.1 pi.hole\n; (1 server found)\n;; global options: +cmd\n;; Got answer:\n;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 4783\n;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1\n\n;; OPT PSEUDOSECTION:\n; EDNS: version: 0, flags:; udp: 4096\n;; QUESTION SECTION:\n;pi.hole.\t\t\tIN\tA\n\n;; ANSWER SECTION:\npi.hole.\t\t2\tIN\tA\t192.168.1.3\n\n;; Query time: 0 msec\n;; SERVER: 127.0.0.1#53(127.0.0.1)\n;; WHEN: Thu Apr 04 04:06:15 UTC 2019\n;; MSG SIZE  rcvd: 52\n\n"
                    },
                    {
                        "Start": "2019-04-04T04:06:45.42666995Z",
                        "End": "2019-04-04T04:06:45.551392086Z",
                        "ExitCode": 0,
                        "Output": "\n; <<>> DiG 9.10.3-P4-Debian <<>> @127.0.0.1 pi.hole\n; (1 server found)\n;; global options: +cmd\n;; Got answer:\n;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 1538\n;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1\n\n;; OPT PSEUDOSECTION:\n; EDNS: version: 0, flags:; udp: 4096\n;; QUESTION SECTION:\n;pi.hole.\t\t\tIN\tA\n\n;; ANSWER SECTION:\npi.hole.\t\t2\tIN\tA\t192.168.1.3\n\n;; Query time: 0 msec\n;; SERVER: 127.0.0.1#53(127.0.0.1)\n;; WHEN: Thu Apr 04 04:06:45 UTC 2019\n;; MSG SIZE  rcvd: 52\n\n"
                    },
                    {
                        "Start": "2019-04-04T04:07:15.553726393Z",
                        "End": "2019-04-04T04:07:15.683036057Z",
                        "ExitCode": 0,
                        "Output": "\n; <<>> DiG 9.10.3-P4-Debian <<>> @127.0.0.1 pi.hole\n; (1 server found)\n;; global options: +cmd\n;; Got answer:\n;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 59106\n;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1\n\n;; OPT PSEUDOSECTION:\n; EDNS: version: 0, flags:; udp: 4096\n;; QUESTION SECTION:\n;pi.hole.\t\t\tIN\tA\n\n;; ANSWER SECTION:\npi.hole.\t\t2\tIN\tA\t192.168.1.3\n\n;; Query time: 0 msec\n;; SERVER: 127.0.0.1#53(127.0.0.1)\n;; WHEN: Thu Apr 04 04:07:15 UTC 2019\n;; MSG SIZE  rcvd: 52\n\n"
                    },
                    {
                        "Start": "2019-04-04T04:07:45.682665769Z",
                        "End": "2019-04-04T04:07:45.813826019Z",
                        "ExitCode": 0,
                        "Output": "\n; <<>> DiG 9.10.3-P4-Debian <<>> @127.0.0.1 pi.hole\n; (1 server found)\n;; global options: +cmd\n;; Got answer:\n;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 6420\n;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1\n\n;; OPT PSEUDOSECTION:\n; EDNS: version: 0, flags:; udp: 4096\n;; QUESTION SECTION:\n;pi.hole.\t\t\tIN\tA\n\n;; ANSWER SECTION:\npi.hole.\t\t2\tIN\tA\t192.168.1.3\n\n;; Query time: 0 msec\n;; SERVER: 127.0.0.1#53(127.0.0.1)\n;; WHEN: Thu Apr 04 04:07:45 UTC 2019\n;; MSG SIZE  rcvd: 52\n\n"
                    },
                    {
                        "Start": "2019-04-04T04:08:15.816422685Z",
                        "End": "2019-04-04T04:08:15.93732397Z",
                        "ExitCode": 0,
                        "Output": "\n; <<>> DiG 9.10.3-P4-Debian <<>> @127.0.0.1 pi.hole\n; (1 server found)\n;; global options: +cmd\n;; Got answer:\n;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 33715\n;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1\n\n;; OPT PSEUDOSECTION:\n; EDNS: version: 0, flags:; udp: 4096\n;; QUESTION SECTION:\n;pi.hole.\t\t\tIN\tA\n\n;; ANSWER SECTION:\npi.hole.\t\t2\tIN\tA\t192.168.1.3\n\n;; Query time: 0 msec\n;; SERVER: 127.0.0.1#53(127.0.0.1)\n;; WHEN: Thu Apr 04 04:08:15 UTC 2019\n;; MSG SIZE  rcvd: 52\n\n"
                    }
                ]
            }
        },
        "Image": "sha256:fafd90299bac38231e04481b43eb0d294ff291c8ea77fe28c936a51b919733de",
        "ResolvConfPath": "/var/lib/docker/containers/ce1c373db763d38d872811cab3ee5cdec1aabe50199f44ac043862ae3e34c4b1/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/ce1c373db763d38d872811cab3ee5cdec1aabe50199f44ac043862ae3e34c4b1/hostname",
        "HostsPath": "/var/lib/docker/containers/ce1c373db763d38d872811cab3ee5cdec1aabe50199f44ac043862ae3e34c4b1/hosts",
        "LogPath": "/var/lib/docker/containers/ce1c373db763d38d872811cab3ee5cdec1aabe50199f44ac043862ae3e34c4b1/ce1c373db763d38d872811cab3ee5cdec1aabe50199f44ac043862ae3e34c4b1-json.log",
        "Name": "/pihole-suffix",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "/Users/admin/pihole/:/etc/pihole/"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {
                "443/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "443"
                    }
                ],
                "53/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "53"
                    }
                ],
                "53/udp": [
                    {
                        "HostIp": "",
                        "HostPort": "53"
                    }
                ],
                "80/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "80"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "unless-stopped",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": [
                "NET_ADMIN"
            ],
            "CapDrop": null,
            "Dns": [
                "127.0.0.1",
                "1.1.1.1"
            ],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "shareable",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DiskQuota": 0,
            "KernelMemory": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": 0,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/c05553a9f2761d7e7bc2e9dcc87612db79f49fc2734cc10c788f616925c43261-init/diff:/var/lib/docker/overlay2/b5d891ba9f4280c3eb8b2e46e92f496de99fdc04b7cc08e9273f0195286cf451/diff:/var/lib/docker/overlay2/df0228dde9fbdfa38194c3fa5c1c64c198a259c58b6e12587e9eccb5bdc0c374/diff:/var/lib/docker/overlay2/3b57230cba4733a7453248c57b1889b1b1d2643423916e8fead6e4e13eadb62f/diff:/var/lib/docker/overlay2/9800e15f9cce2ea34e2aaf264f45453e748990e4cd93f86d97d564d25ce57712/diff:/var/lib/docker/overlay2/fb055cb5b655b04d88119f3d8f4779a71b2af9c612ea05cf84c700d21477b43b/diff:/var/lib/docker/overlay2/5151575ae6a4f1e1ec7a1e237daeb33bb49b3a9db4ddb6fe5b3434e63bab007a/diff:/var/lib/docker/overlay2/0609e6f9c1fee35f366827ecdeb361bcb378ca93abc5e69ae2682a9d9f73c6f6/diff:/var/lib/docker/overlay2/37e1d302069da91bc2750ebfd38ee5e39dc44e36da58f62052060f605ecdef30/diff:/var/lib/docker/overlay2/28661db973230504dbef9613c435d6eb47e5be005371b6bbdd9f7a9a5d0a4e43/diff:/var/lib/docker/overlay2/307363af9c467480ba44499e7ed192c96e28074f031cab48395cca474524dabc/diff:/var/lib/docker/overlay2/477c7972b5228bbe0dc910e555bdef20a9890b15109577fb30a8a60da8ad4481/diff",
                "MergedDir": "/var/lib/docker/overlay2/c05553a9f2761d7e7bc2e9dcc87612db79f49fc2734cc10c788f616925c43261/merged",
                "UpperDir": "/var/lib/docker/overlay2/c05553a9f2761d7e7bc2e9dcc87612db79f49fc2734cc10c788f616925c43261/diff",
                "WorkDir": "/var/lib/docker/overlay2/c05553a9f2761d7e7bc2e9dcc87612db79f49fc2734cc10c788f616925c43261/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/Users/admin/pihole",
                "Destination": "/etc/pihole",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "ab3e97a8102c",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "443/tcp": {},
                "53/tcp": {},
                "53/udp": {},
                "67/udp": {},
                "80/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "WEBPASSWORD=Password",
                "DNS1=1.1.1.1",
                "DNS2=1.0.0.1",
                "ServerIP=192.168.1.3",
                "PATH=/opt/pihole:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "S6OVERLAY_RELEASE=https://github.com/just-containers/s6-overlay/releases/download/v1.21.7.0/s6-overlay-amd64.tar.gz",
                "PIHOLE_INSTALL=/root/ph_install.sh",
                "PHP_ENV_CONFIG=/etc/lighttpd/conf-enabled/15-fastcgi-php.conf",
                "PHP_ERROR_LOG=/var/log/lighttpd/error.log",
                "IPv6=True",
                "S6_LOGGING=0",
                "S6_KEEP_ENV=1",
                "S6_BEHAVIOUR_IF_STAGE2_FAILS=2",
                "FTL_CMD=no-daemon",
                "DNSMASQ_USER=root",
                "VERSION=v4.2.2",
                "ARCH=amd64"
            ],
            "Cmd": null,
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "dig @127.0.0.1 pi.hole || exit 1"
                ]
            },
            "ArgsEscaped": true,
            "Image": "pihole/pihole:latest",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/s6-init"
            ],
            "OnBuild": null,
            "Labels": {
                "image": "pihole/pihole:v4.2.2_amd64",
                "maintainer": "adam@diginc.us",
                "url": "https://www.github.com/pi-hole/docker-pi-hole"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "2d026215f5944391dd178e32e93f92e085a782c80c2417a05e131378528fed85",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "443/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "443"
                    }
                ],
                "53/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "53"
                    }
                ],
                "53/udp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "53"
                    }
                ],
                "67/udp": null,
                "80/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "80"
                    }
                ]
            },
            "SandboxKey": "/var/run/docker/netns/2d026215f594",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "312b032dfd1d4549ecc5f1da8bbc8534b4e70e70c7b914cc1cb46d1deccbc84b",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.2",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:02",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [],
                    "NetworkID": "25e5cea229aed6ccfaeae322475a357279337582a3b0149504cfb9176eb1c7e0",
                    "EndpointID": "312b032dfd1d4549ecc5f1da8bbc8534b4e70e70c7b914cc1cb46d1deccbc84b",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:11:00:02",
                    "DriverOpts": null
                }
            }
        }
    }
]
suffix:~ admin$
```

**Comment by gmartinerm** [comment]
One last update. If I query via dig the cloudflare instance from 192.168.1.3, here is the output
```
suffix:~ admin$ dig -p 5053 @192.168.1.2 cisco.com

; <<>> DiG 9.8.3-P1 <<>> -p 5053 @192.168.1.2 cisco.com
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 47786
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;cisco.com.                     IN      A

;; ANSWER SECTION:
cisco.com.              3600    IN      A       72.163.4.185

;; Query time: 224 msec
;; SERVER: 192.168.1.2#5053(192.168.1.2)
;; WHEN: Thu Apr  4 00:21:04 2019
;; MSG SIZE  rcvd: 43

suffix:~ admin$ dig -p 5053 @192.168.1.3 cisco.com

; <<>> DiG 9.8.3-P1 <<>> -p 5053 @192.168.1.3 cisco.com
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 9035
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;cisco.com.                     IN      A

;; ANSWER SECTION:
cisco.com.              3581    IN      A       72.163.4.185

;; Query time: 2 msec
;; SERVER: 192.168.1.3#5053(192.168.1.3)
;; WHEN: Thu Apr  4 00:21:07 2019
;; MSG SIZE  rcvd: 43

suffix:~ admin$

suffix:~ admin$ dig -p 5053 @127.0.0.1 cisco.com

; <<>> DiG 9.8.3-P1 <<>> -p 5053 @127.0.0.1 cisco.com
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 7623
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;cisco.com.                     IN      A

;; ANSWER SECTION:
cisco.com.              3492    IN      A       72.163.4.185

;; Query time: 1 msec
;; SERVER: 127.0.0.1#5053(127.0.0.1)
;; WHEN: Thu Apr  4 00:22:36 2019
;; MSG SIZE  rcvd: 43

suffix:~ admin$

```

**Comment by qdm12** [comment]
That is odd indeed. A few questions just to be sure:
1. The cloudflare DNS works (if dig works), right?
1. Pihole can't communicate with it locally on your Mac mini on its published port 5053/udp, right?
1. Pihole on your NAS can use the cloudflare DNS container of the Mac mini on port 5053,right?

Thanks

**Comment by gmartinerm** [comment]
I found a workaround. As long as I specify the the IP address of my mac mini and not the loopback the resolution works.  One quick question why this is happening?   time servers hosted by nist.gov do not resolve

```
suffix:~ admin$ dig -p 5053 @192.168.1.3  time-a-g.nist.gov

; <<>> DiG 9.8.3-P1 <<>> -p 5053 @192.168.1.3 time-a-g.nist.gov
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 43797
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;time-a-g.nist.gov.             IN      A

;; Query time: 1 msec
;; SERVER: 192.168.1.3#5053(192.168.1.3)
;; WHEN: Mon Apr 15 23:34:47 2019
;; MSG SIZE  rcvd: 35

suffix:~ admin$ dig -p 5053 @192.168.1.2  time-a-g.nist.gov

; <<>> DiG 9.8.3-P1 <<>> -p 5053 @192.168.1.2 time-a-g.nist.gov
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 40354
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;time-a-g.nist.gov.             IN      A

;; Query time: 1 msec
;; SERVER: 192.168.1.2#5053(192.168.1.2)
;; WHEN: Mon Apr 15 23:34:54 2019
;; MSG SIZE  rcvd: 35

suffix:~ admin$
```
Any clue?

**Comment by gmartinerm** [comment]
Forgot to post the direct query without going through my local servers
```
suffix:~ admin$ dig @8.8.8.8  time-a-g.nist.gov

; <<>> DiG 9.8.3-P1 <<>> @8.8.8.8 time-a-g.nist.gov
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 51386
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;time-a-g.nist.gov.             IN      A

;; ANSWER SECTION:
time-a-g.nist.gov.      568     IN      A       129.6.15.28

;; Query time: 4 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: Mon Apr 15 23:37:54 2019
;; MSG SIZE  rcvd: 51

suffix:~ admin$

```

**Comment by gmartinerm** [comment]
I have an incidence open with pihole as well. See the following
```
https://github.com/pi-hole/pi-hole/issues/2708
```

**Comment by gmartinerm** [comment]
None of the US goverment websites seems to resolve
```
suffix:~ admin$ dig -p 5053 @192.168.1.3 time-b-g.nist.gov

; <<>> DiG 9.8.3-P1 <<>> -p 5053 @192.168.1.3 time-b-g.nist.gov
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 64234
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;time-b-g.nist.gov.             IN      A

;; Query time: 1 msec
;; SERVER: 192.168.1.3#5053(192.168.1.3)
;; WHEN: Tue Apr 16 05:51:36 2019
;; MSG SIZE  rcvd: 35

suffix:~ admin$  top
suffix:~ admin$ dig -p 5053 @192.168.1.3 www.senate.gov

; <<>> DiG 9.8.3-P1 <<>> -p 5053 @192.168.1.3 www.senate.gov
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 33423
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;www.senate.gov.                        IN      A

;; Query time: 1 msec
;; SERVER: 192.168.1.3#5053(192.168.1.3)
;; WHEN: Tue Apr 16 05:57:00 2019
;; MSG SIZE  rcvd: 32

suffix:~ admin$ dig -p 5053 @192.168.1.3 house.senate.gov

; <<>> DiG 9.8.3-P1 <<>> -p 5053 @192.168.1.3 house.senate.gov
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 37642
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;house.senate.gov.              IN      A

;; Query time: 1 msec
;; SERVER: 192.168.1.3#5053(192.168.1.3)
;; WHEN: Tue Apr 16 05:57:08 2019
;; MSG SIZE  rcvd: 34

suffix:~ admin$ dig -p 5053 @192.168.1.3 whitehouse.senate.gov

; <<>> DiG 9.8.3-P1 <<>> -p 5053 @192.168.1.3 whitehouse.senate.gov
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 44667
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;whitehouse.senate.gov.         IN      A

;; Query time: 1 msec
;; SERVER: 192.168.1.3#5053(192.168.1.3)
;; WHEN: Tue Apr 16 05:57:15 2019
;; MSG SIZE  rcvd: 39
```

**Comment by qdm12** [comment]
That's because some US government websites are considered malicious, it's in the [malicious block list](https://raw.githubusercontent.com/qdm12/updated/master/files/malicious-hostnames.updated). I am in the process of separating these malicious hostnames in different files as it's indeed annoying to have US gov websites blocked sometimes. I will update here once this is done.

**Comment by qdm12** [comment]
Hi @gmartinerm,

I've added two new environment variables to solve that problem.
1. `BLOCK_NSA` can be set to `off` (default) or `on` and will block some US government websites as they were reported as spying on you.
1. `UNBLOCK` allows you to unblock specific hostnames with a comma separated string such as `website1.gov,website2.com`. That can be useful to keep the `BLOCK_NSA=on` but allow some specific US government websites you need.

Feel free to re-open or create another issue if there is still something wrong.

Happy resolving !
